mergify_engine/signals.py

```diff
@@ -59,9 +102,10 @@ class SignalBase(abc.ABC):
     @abc.abstractmethod
     async def __call__(
         self,
-        ctxt: context.Context,
+        repository: "context.Repository",
+        pull_request: github_types.GitHubPullRequestNumber,
```

Comment:
What about passing the whole PR structure?

Reply:
For now, we track only think where we have the full context, but when I will add signals in Train and TrainCar, we may not have them.